src/app/[locale]/seller/registration/page.tsx

```diff
@@ -57,59 +57,55 @@ const SellerRegistrationForm = () => {
   const [linkUrl, setLinkUrl] = useState('');
 
   useEffect(() => {
-    const fetchSellCenter = async () => {
+    const fetchData = async () => {
+      if (!currentUser) {
+        logger.info('User not logged in; attempting auto-login..');
+        autoLoginUser();
+        return;
+      }
+  
       try {
-        const mapCenterData = await fetchMapCenter(); // Function to fetch map center from backend
-        if (mapCenterData) {
-          const { longitude, latitude } = mapCenterData;
-          if (longitude !== undefined && latitude !== undefined) {
-            setSellCenter({ lng: longitude, lat: latitude });
+        // Fetch Sell Center
+        if (!sellCenter) {
+          const mapCenterData = await fetchMapCenter();
+          if (mapCenterData) {
+            const { longitude, latitude } = mapCenterData;
+            if (latitude !== undefined && longitude !== undefined) {
+              setSellCenter({ lng: longitude, lat: latitude });
+            }
           }
         }
-      } catch (error) {
-        logger.error('Error fetching sellCenter from backend:', error);
-      }
-    };
-
-    fetchSellCenter();
-  }, [currentUser]);
-
-  // Fetch seller data and user settings on component mount
-  useEffect(() => {
-    if (!currentUser) {
-      logger.info("User not logged in; attempting auto-login..");
-      autoLoginUser();
-    }
-
-    const getSellerData = async () => {
-      try {
-        const data = await fetchSellerRegistration();
-        if (data) {
-          setDbSeller(data);
-        } else {
-          setDbSeller(null);
+  
+        // Fetch Seller Data
+        if (!dbSeller) {
```

Comment:
this logic will work fine at initial screen loading but the subsequent updated value may not be displayed because dbSeller != null